e2e-tests/cypress/tests/integration/channels/notifications/reset_notification_when_settings_canceled_spec.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channels @notifications
+import {callsPlugin} from '../../../utils/plugins';
+
+describe('Notifications', () => {
+    before(() => {
+        // # Upload and enable Calls plugin
+        cy.shouldNotRunOnCloudEdition();
+        cy.shouldHavePluginUploadEnabled();
+        cy.apiUploadAndEnablePlugin(callsPlugin);
+
+        cy.apiInitSetup().then(({team, user, channel}) => {
+            // # Login as user and visit channel
+            cy.apiLogin(user);
+            cy.visit(`/${team.name}/channels/${channel.name}`);
+        });
+    });
+
+    it('MM-53166 Notification sound modal selection should reset when settings canceled', () => {
+        // # Call function that clicks on Settings -> Notifications -> Desktop Notifications -> Notification sound -> Change sound -> Cancel -> Desktop Notifications
+        openSettingsAndChangeNotification('desktopNotification');
+
+        // # Call function that clicks on Settings -> Notifications -> Desktop Notifications -> Notification sound for incoming calls -> Cancel -> Desktop Notifications
+        openSettingsAndChangeNotification('callsDesktopSound');
+    });
+
+    function openSettingsAndChangeNotification(type) {
+        // # Open 'Settings' modal
+        cy.uiOpenSettingsModal().within(() => {
+            // # Navigate to Desktop Notification Settings
+            navigateToDesktopNotificationSettings(type);
+
+            // # Change Notification selection
+            setNotificationToDown(type);
+
+            // # Click Cancel button
+            cy.uiCancelButton().click();
+
+            // # Navigate to Desktop Notification Settings
+            navigateToDesktopNotificationSettings(type);
+        });
+    }
+
+    function setNotificationToDown(type) {
+        switch (type) {
+        case 'desktopNotification':
+            // # Change Notification sound selection value is set to Down
+            cy.get('#displaySoundNotification').
+                find('.react-select__dropdown-indicator').
+                click().
+                get('.react-select__menu').
+                contains('Down').
+                click();
```

Comment:
57-62

can you try something like this
```
cy.get('#displaySoundNotification').should('have.text', 'Bing').should('be.enabled').click();
cy.findbyText('Down').click()
cy.get('#displaySoundNotification').should('have.text', 'Down').
...rest of the code
```
was just curious if we can avoid chaining and using class, (you can add attribute to element like `data-testid`, if they are not directly getting located.)

Reply:
I tried your suggestion. Unfortunately, I got error like this
```'<div#displaySoundNotification.react-select.notification-sound-dropdown.css-2b097c-container>' to have text 'Down', but the text was ' option Down, selected.   6 results available. Select is focused , press Down to open the menu, Down'```

I tried to search other spec file in this project but I can't find that tested react-select. So I just follow react-select e2e test sample here: https://github.com/JedWatson/react-select/blob/master/cypress/integration/single-select.spec.ts

Reply:
I also tried
```
cy.get('#displaySoundNotification').click()
cy.contains('Down').click()
```

But i got error stated
```
Timed out retrying after 20050ms: cy.click() failed because this element:



<p id="aria-context">option ...</p>



is being covered by another element:



<div class="react-select__value-container react-select__value-container--has-value css-1hwfws3">...</div>
```

Reply:
Sure, I fix it in my new commit.